src/modules/Elsa.Alterations/Services/AlterationPlanManager.cs

```diff
@@ -0,0 +1,43 @@
+using Elsa.Alterations.Core.Contracts;
+using Elsa.Alterations.Core.Entities;
+using Elsa.Alterations.Core.Enums;
+using Elsa.Alterations.Core.Filters;
+using Elsa.Alterations.Core.Notifications;
+using Elsa.Common.Contracts;
+using Elsa.Mediator.Contracts;
+
+namespace Elsa.Alterations.Services;
+
+/// <inheritdoc />
+public class AlterationPlanManager(IAlterationPlanStore planStore, IAlterationJobStore jobStore, ISystemClock systemClock, INotificationSender notificationSender) : IAlterationPlanManager
+{
+    /// <inheritdoc />
+    public async Task<AlterationPlan?> GetPlanAsync(string planId, CancellationToken cancellationToken = default)
+    {
+        var planFilter = new AlterationPlanFilter { Id = planId };
+        return await planStore.FindAsync(planFilter, cancellationToken);
+    }
+
+    /// <inheritdoc />
+    public async Task<bool> GetIsAllJobsCompletedAsync(string planId, CancellationToken cancellationToken = default)
+    {
+        // Check if all jobs are completed.
+        var jobFilter = new AlterationJobFilter
```

Comment:
What if the jobs do not take much time and some of them are already completed?

Reply:
Or failed?

Reply:
f some jobs have completed while others are still running, then this method will return false, since not all jobs have completed yet, given that there is a positive number of jobs with a status of Running or Pending.

But I am probably missing your point, so please elaborate if I did.

Reply:
Nevermind, I read this completely the wrong way.